src/main/java/com/emc/mongoose/storage/driver/pravega/PravegaStorageDriver.java

```diff
@@ -163,6 +167,65 @@ protected boolean submit(O op) throws InterruptRunException, IllegalStateExcepti
         return false;
     }
 
+    protected boolean submitItemCreate(
+        final DataOperation<? extends DataItem> eventOperation, final DataItem eventItem
+    )
+    {
+        final long eventSize;
+        try {
+            eventSize = eventItem.size();
+        } catch(final IOException e) {
+            throw new AssertionError(e);
+        }
+
+
+        final String path = eventOperation.dstPath();
+        final String scope = path.substring(0, path.indexOf("/"));
+        final String streamName = path.substring(path.indexOf("/")+1);
+
+        if (scopeMap.get(scope).get(streamName) == null) {
+            Loggers.ERR.debug(
+                    "Failed to create event: the stream {} not found in the scope {}", streamName, scope);
+            eventOperation.status(Operation.Status.RESP_FAIL_UNKNOWN);
+            return false;
+        };
+
+        StreamConfiguration streamConfig = StreamConfiguration.builder()
+                .scalingPolicy(ScalingPolicy.fixed(1))
+                .build();
+
+        streamManager.createStream(scope, streamName, streamConfig);
+
+        try (ClientFactory clientFactory = ClientFactory.withScope(scope, URI.create(uriSchema));
+             EventStreamWriter<ByteBuffer> writer = clientFactory.createEventWriter(streamName,
+                     null,
+                     EventWriterConfig.builder().build())) {
+            final String routingKey = null;
+            final DataInput dataInput = eventItem.dataInput();
+            final ByteBuffer outBuff = dataInput.getLayer(eventItem.layer());
+
+            final CompletableFuture writeFuture = writer.writeEvent(outBuff);
```

Comment:
need to set the offset to the output buffer before
the offset may be determined by calling eventItem.offset() method
also, it's necessary to set the output buffer limit to make it writing no more than the eventItem.size() bytes